pkg/controller/installation/products/threescale/reconciler.go

```diff
@@ -356,25 +363,31 @@ func (r *Reconciler) reconcileRHSSOIntegration(ctx context.Context, serverClient
 	if err != nil {
 		return v1alpha1.PhaseFailed, err
 	}
-	site := rhssoConfig.GetHost() + "/auth/realms/" + rhssoRealm
-	res, err := r.tsClient.AddSSOIntegration(map[string]string{
-		"kind":                              "keycloak",
-		"name":                              "rhsso",
-		"client_id":                         clientId,
-		"client_secret":                     clientSecret,
-		"site":                              site,
-		"skip_ssl_certificate_verification": "true",
-		"published":                         "true",
-	}, *accessToken)
-
-	if err != nil || res.StatusCode != http.StatusCreated && res.StatusCode != http.StatusUnprocessableEntity {
+
+	_, err = r.tsClient.GetAuthenticationProviderByName(rhssoIntegrationName, *accessToken)
+	if err != nil && !tsIsNotFoundError(err) {
 		return v1alpha1.PhaseFailed, err
 	}
+	if tsIsNotFoundError(err) {
+		site := rhssoConfig.GetHost() + "/auth/realms/" + rhssoRealm
+		res, err := r.tsClient.AddAuthenticationProvider(map[string]string{
+			"kind":                              "keycloak",
+			"name":                              rhssoIntegrationName,
+			"client_id":                         clientId,
+			"client_secret":                     clientSecret,
+			"site":                              site,
+			"skip_ssl_certificate_verification": "true",
```

Comment:
This should probably use `selfSignedCerts` value from InstallationSpec.

Reply:
That value is set on the http client:

https://github.com/integr8ly/integreatly-operator/blob/38c490ff568f2ad201c692136cad6d9ddae1fd3e/pkg/controller/installation/products/reconciler.go#L69-L75